GitLab Pipelines Edit (#20132)
DataDog/documentation: content/en/continuous_integration/pipelines/gitlab.md

## Enable job log collection

The following GitLab versions support collecting job logs:

* GitLab.com (SaaS)
* GitLab >= 14.8 (self-hosted) only if you are using [object storage to store job logs][7]

To enable collection of job logs:

> 1. Enable the `datadog_integration_logs_collection` [feature flag][8] in your GitLab self-hosted or GitLab.com account. This allows you to see the **Enable job logs collection** button on the [Pipeline Setup page][17].

Reviewer: ```suggestion
1. Enable the `datadog_integration_logs_collection` [feature flag][8] in your GitLab self-hosted or GitLab.com account. This allows you to see the **Enable job logs collection** check box on the [Pipeline Setup page][17].
```

> 1. Enable the `datadog_integration_logs_collection` [feature flag][8] in your GitLab self-hosted or GitLab.com account. This allows you to see the **Enable job logs collection** check box on the [Pipeline Setup page][17].

Reviewer 2: ```suggestion
1. Enable the `datadog_integration_logs_collection` [feature flag][8] in your GitLab self-hosted or GitLab.com account. This allows you to see the **Enable job logs collection** checkbox on the [Pipeline Setup page][17].
```

Reviewer 2: `checkbox` looks to be much more frequently used in the documentation site. This is also preferred in some style guides:
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/check-checkbox-check-mark

2. Click **Enable job logs collection** and click **Save changes**.

> Job logs are collected in [Log Management][9] and are automatically correlated with the GitLab pipeline in CI Visibility. Log files larger than 1GiB are truncated.

Reviewer 2: ```suggestion
Job logs are collected in [Log Management][9] and are automatically correlated with the GitLab pipeline in CI Visibility. Log files larger than one GiB are truncated.
```

> Job logs are collected in [Log Management][9] and are automatically correlated with the [GitLab pipeline][18] in CI Visibility. Log files larger than 1GiB are truncated.

Author: ```suggestion
Job logs are collected in [Log Management][9] and are automatically correlated with the GitLab pipeline in CI Visibility. Log files larger than 1GiB are truncated.
```

<div class="alert alert-info"><strong>Note</strong>: Logs are billed separately from CI Visibility.</div>

> For more information about processing job logs collected from the GitLab integration, see the [Processors documentation][19].

Reviewer 2: ```suggestion
For more information about processing job logs collected from the GitLab integration, see the [Processors documentation][18].
```

[14]: /continuous_integration/pipelines/gitlab/?tab=gitlabcom#correlate-infrastructure-metrics-to-jobs
[15]: /continuous_integration/pipelines/gitlab/?tab=gitlabcom#view-error-messages-for-pipeline-failures
[16]: /account_management/teams/
[17]: https://app.datadoghq.com/ci/setup/pipeline?provider=gitlab

> [19]: /logs/log_configuration/processors/

Reviewer 2: ```suggestion
[18]: /logs/log_configuration/processors/
```

[14]: /continuous_integration/pipelines/gitlab/?tab=gitlabcom#correlate-infrastructure-metrics-to-jobs
[15]: /continuous_integration/pipelines/gitlab/?tab=gitlabcom#view-error-messages-for-pipeline-failures
[16]: /account_management/teams/
[17]: https://app.datadoghq.com/ci/setup/pipeline?provider=gitlab

> [18]: https://app.datadoghq.com/ci/pipeline-executions?query=ci_level%3Apipeline%20%40ci.provider.name%3Agitlab&index=cipipeline

Reviewer: This won't get the user to anything where they can see logs or pipelines that have logs exclusively. I don't think we have a deep link for this

Author: ```suggestion
```
